storagenode/pieces/store.go

```diff
@@ -312,6 +312,27 @@ func (store *Store) Trash(ctx context.Context, satellite storj.NodeID, pieceID s
 	return Error.Wrap(err)
 }
 
+// EmptyTrash deletes pieces in the trash that have been in there longer than trashExpiryInterval
+func (store *Store) EmptyTrash(ctx context.Context, satelliteID storj.NodeID, trashedBefore time.Time) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	deletedIDs, err := store.blobs.EmptyTrash(ctx, satelliteID[:], trashedBefore)
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	var pieceID storj.PieceID
+	for _, deletedID := range deletedIDs {
+		pieceID, err = storj.PieceIDFromBytes(deletedID)
```

Comment:
This is overwriting err to nil every time, which misses the collected `DeleteExpiration` errors.

Reply:
Done.